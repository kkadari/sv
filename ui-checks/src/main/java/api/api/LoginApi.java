package api.api;

import io.restassured.response.Response;

import static io.restassured.RestAssured.given;

public class LoginApi {

    public static Response login(String baseUrl, String username, String password){
        return given()
                .params("username", username, "password", password)
               .when()
                .post(baseUrl + "/cs_login");
    }
}
