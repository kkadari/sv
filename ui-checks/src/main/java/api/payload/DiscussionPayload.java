package api.payload;

public class DiscussionPayload {

//    {
//        "subject": "asdasdasads",
//        "markAsQuestion": false,
//        "body": "<body><p>asdasdasdasdasasdas</p></body>",
//        "handlingLevel": "2",
//        "publishBar": {
//            "visibility": "all"
//        },
//        "notAttributable": false,
//    }

    private String subject;
    private Boolean markAsQuestion;
    private String body;
    private String handlingLevel;
    private PublishPayload publishBar;
    private boolean notAttributable;

    public DiscussionPayload(String subject, Boolean markAsQuestion, String body, String handlingLevel, boolean notAttributable) {
        this.subject = subject;
        this.markAsQuestion = markAsQuestion;
        this.body = body;
        this.handlingLevel = handlingLevel;
        this.publishBar = new PublishPayload();
        this.notAttributable = notAttributable;
    }

    public String getSubject() {
        return subject;
    }

    public Boolean getMarkAsQuestion() {
        return markAsQuestion;
    }

    public String getBody() {
        return body;
    }

    public String getHandlingLevel() {
        return handlingLevel;
    }

    public PublishPayload getPublishBar() {
        return publishBar;
    }

    public boolean isNotAttributable() {
        return notAttributable;
    }
}
