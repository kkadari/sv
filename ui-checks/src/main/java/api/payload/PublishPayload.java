package api.payload;

public class PublishPayload {

    private String visibility;

    public PublishPayload() {
        this.visibility = "all";
    }

    public String getVisibility() {
        return visibility;
    }
}
