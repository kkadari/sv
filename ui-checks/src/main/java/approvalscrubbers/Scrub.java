package approvalscrubbers;

import java.util.ArrayList;

public class Scrub {

    public static String scrub(ArrayList<String> scrubbers, String content){
        for(String scrubber : scrubbers){
            content = content.replaceAll(scrubber, "***scrubbed***");
        }

        return content;
    }

}
