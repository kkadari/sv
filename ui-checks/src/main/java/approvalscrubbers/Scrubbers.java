package approvalscrubbers;

public class Scrubbers {

    public static String scrubJiveId(){
        return "id=\"jive-[0-9]*\"";
    }

    public static String scrubThreadRef(){
        return "\\/thread\\/[0-9]*";
    }

    public static String scrubDisplayObjectKeys(){
        return "name=\"displayedObjectKeys\" value=\"[0-9_,]*\"";
    }

}
