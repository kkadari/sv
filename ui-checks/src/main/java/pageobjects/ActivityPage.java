package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

import java.util.List;

public class ActivityPage extends PageObject {

    @FindBy(how = How.CSS, using = ".j-act-title")
    List<WebElement> divTitles;

    public ActivityPage(WebDriver driver){
        super(driver);
    }

    public String returnStreamListHtml(){
        String streams = "";

        for(WebElement title : divTitles){
            streams += title.getAttribute("outerHTML");
        }

        return streams;
    }

}
