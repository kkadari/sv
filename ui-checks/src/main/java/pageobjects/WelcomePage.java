package pageobjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class WelcomePage extends PageObject {

    @FindBy(how = How.CSS, using = ".content-small .widget-container")
    WebElement divWidgetContainer;

    @FindBy(how = How.CSS, using = "#whats-happening")
    WebElement divWhatsHappening;

    @FindBy(how = How.CSS, using = "#feed-popular")
    WebElement listFeedPopular;

    @FindBy(how = How.CSS, using = "#feed-following")
    WebElement listFeedFollowing;

    @FindBy(how = How.CSS, using = ".event-content")
    WebElement divContent;

    @FindBy(how = How.CSS, using = ".footer-main")
    WebElement divFooter;

    @FindBy(how = How.CSS, using = "#customBrowseMenuAnchor")
    WebElement linkContentMenu;

    @FindBy(how = How.CSS, using = "#customBrowseMenu")
    WebElement divBrowseMenu;

    @FindBy(how = How.CSS, using = "#customBrowseMenu .j-pop-desc")
    WebElement divContentMenuItem;

    @FindBy(how = How.CSS, using = "#customMembersMenuAnchor")
    WebElement linkMembersMenu;

    @FindBy(how = How.CSS, using = "#customMembersMenu")
    WebElement divMembersMenu;

    @FindBy(how = How.CSS, using = "#navCommunitiesAndNodes")
    WebElement linkCommunityMenu;

    @FindBy(how = How.CSS, using = "#menuCommunitiesAndNodes")
    WebElement divCommunityMenu;

    @FindBy(how = How.CSS, using = "#menuCommunitiesAndNodes .j-pop-desc")
    WebElement divCommunityMenuItem;

    private WebDriverWait wait = new WebDriverWait(driver, timeout);

    public WelcomePage(WebDriver driver){
        super(driver);
    }

    public String returnWidgetContainerHtml(){
        return divWidgetContainer.getAttribute("outerHTML");
    }

    public String returnWhatsHappeningHtml(){
        wait.until(ExpectedConditions.visibilityOf(divContent));

        return divWhatsHappening.getAttribute("outerHTML");
    }

    public void clickFeedPopular() {
        wait.until(ExpectedConditions.visibilityOf(listFeedPopular));

        listFeedPopular.click();
    }

    public void clickFeedFollowing() {
        wait.until(ExpectedConditions.visibilityOf(listFeedFollowing));

        listFeedFollowing.click();
    }

    public String returnFooterHtml() {
        return divFooter.getAttribute("outerHTML");
    }

    public void clickContent() {
        linkContentMenu.click();
    }

    public String returnContentMenuPopupHtml(){
        wait.until(ExpectedConditions.visibilityOf(divContentMenuItem));

        return divBrowseMenu.getAttribute("outerHTML");
    }

    public void clickMembers() {
        linkMembersMenu.click();
    }

    public String returnMembersMenuPopupHtml() {
        return divMembersMenu.getAttribute("outerHTML");
    }

    public void clickCommunityAndNodes() {
        linkCommunityMenu.click();
    }

    public String returnCommunityAndNodesPopupHtml() {
        wait.until(ExpectedConditions.visibilityOf(divCommunityMenuItem));

        return divCommunityMenu.getAttribute("outerHTML");
    }

}
