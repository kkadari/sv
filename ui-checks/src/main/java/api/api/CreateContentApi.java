package api.api;

import api.payload.DiscussionPayload;
import io.restassured.http.ContentType;
import io.restassured.http.Cookies;

import static io.restassured.RestAssured.given;

public class CreateContentApi {

    public static void createDiscussion(Cookies cookies, String baseUrl, DiscussionPayload discussionPayload){
        given()
            .contentType(ContentType.JSON)
            .cookies(cookies)
            .body(discussionPayload)
        .when()
            .post(baseUrl + "/__services/v2/rest/discussion");
    }
}
