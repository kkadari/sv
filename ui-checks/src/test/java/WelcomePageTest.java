import api.api.CreateContentApi;
import api.payload.DiscussionPayload;
import approvalscrubbers.Scrub;
import approvalscrubbers.Scrubbers;
import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;
import pageobjects.WelcomePage;

import java.util.ArrayList;

public class WelcomePageTest extends TestSetup {

    @Before
    public void generateTestContent(){
        for(int i = 1; i <= 7; i++){
            DiscussionPayload discussionPayload = new DiscussionPayload("Check discussion", false, "Generated for automation", "1", false);

            CreateContentApi.createDiscussion(cookies, baseUrl, discussionPayload);
        }
    }

    @Test
    public void approveStaticCallToAction(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        String rawWidgetContainer = welcome.returnWidgetContainerHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubJiveId());
        }};

        String scrubbedWidgetContainer = Scrub.scrub(scrubbers, rawWidgetContainer);

        Approvals.verifyHtml(scrubbedWidgetContainer);
    }

    @Test
    public void approveWhatsHappening(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        String whatsHappeningContainer = welcome.returnWhatsHappeningHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubJiveId());
            add(Scrubbers.scrubThreadRef());
            add(Scrubbers.scrubDisplayObjectKeys());
        }};

        String scrubbedWhatsHappeningContainer = Scrub.scrub(scrubbers, whatsHappeningContainer);

        Approvals.verifyHtml(scrubbedWhatsHappeningContainer);
    }

    @Test
    public void approveWhatsHappeningPopularTab(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        welcome.clickFeedPopular();

        String whatsHappeningContainer = welcome.returnWhatsHappeningHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubJiveId());
            add(Scrubbers.scrubThreadRef());
            add(Scrubbers.scrubDisplayObjectKeys());
        }};

        String scrubbedWhatsHappeningContainer = Scrub.scrub(scrubbers, whatsHappeningContainer);

        Approvals.verifyHtml(scrubbedWhatsHappeningContainer);
    }

    @Test
    public void approveWhatsHappeningFollowingTab(){
        driver.navigate().to(baseUrl + "/welcome");

        WelcomePage welcome = new WelcomePage(driver);
        welcome.clickFeedFollowing();

        String whatsHappeningContainer = welcome.returnWhatsHappeningHtml();

        ArrayList<String> scrubbers = new ArrayList<String>(){{
            add(Scrubbers.scrubJiveId());
            add(Scrubbers.scrubThreadRef());
            add(Scrubbers.scrubDisplayObjectKeys());
        }};

        String scrubbedWhatsHappeningContainer = Scrub.scrub(scrubbers, whatsHappeningContainer);

        Approvals.verifyHtml(scrubbedWhatsHappeningContainer);
    }

}
