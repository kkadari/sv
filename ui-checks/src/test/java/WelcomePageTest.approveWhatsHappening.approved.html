<div id="whats-happening">
		<h2>What's Happening <a class="pull-right view-feeds" href="/activity">view feeds</a></h2>

		<ul class="feeds">
			<li class="active" id="feed-all">All</li>
			<li id="feed-popular">Popular</li>			
			<li id="feed-following">Following</li>
		</ul>
	
		<ul class="events">

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>

	<!-- WhatsHappeningContentObject -->

    <li class="newpost">
    
      <img class="icon" src="/custom-images/overview/newpost.png">
	  <div class="event-content">
      	<p>New post "<a href="***scrubbed***">Check discussion</a>"</p>
      	<p>
by
<a href="/people/admin" data-externalid="" data-username="admin" data-avatarid="-1" ***scrubbed*** data-userid="1" data-presence="null" class="jiveTT-hover-user jive-username-link">admin</a>      		
      	</p>
     </div>
      
      	<span class="ihm-pill ihm-browse-1 tlp level-1" title="RED: May be shared with named participant(s) only.">
Red      	</span>
    </li>
		</ul>
	
    	<div class="more-status-content">
        	<span class="jive-whats-new-more">
            	<a href="#" class="jive-icon-link-forward-content jive-more-content">More</a>   
        	</span>
    	</div>

		<form class="whats-happening-ajax-form" action="/whats-happening-ajax.jspa">
    		<input type="hidden" name="containerID" value="1">
    		<input type="hidden" name="containerType" value="14">
			<input type="hidden" ***scrubbed***>
		</form>
	</div>
